commons/all/src/main/java/org/infinispan/commons/util/SslContextFactory.java

```diff
@@ -112,6 +122,13 @@ public SslContextFactory sslProtocol(String sslProtocol) {
       return this;
    }
 
+   public SslContextFactory provider(String provider) {
+      if (provider != null) {
+         this.provider = provider;
```

Comment:
just to clarify, the `getContext` (SSLContext) uses `SSL_PROVIDER` instead of this configured provider. Is it intentional?